aws-iot-thing/aws-iot-thing.json

```diff
@@ -0,0 +1,80 @@
+{
+  "typeName": "AWS::IoT::Thing",
+  "description": "Resource Type definition for AWS::IoT::Thing",
+  "additionalProperties": false,
+  "properties": {
+    "Id": {
+      "type": "string"
+    },
+    "Arn": {
+      "type": "string"
+    },
+    "AttributePayload": {
+      "$ref": "#/definitions/AttributePayload"
+    },
+    "ThingName": {
+      "type": "string",
+      "pattern": "[a-zA-Z0-9:_-]+",
+      "minLength": 1,
+      "maxLength": 128
+    }
+  },
+  "definitions": {
+    "AttributePayload": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "Attributes": {
+          "type": "object",
+          "additionalProperties": false,
+          "patternProperties": {
+            "[a-zA-Z0-9]+": {
+              "type": "string"
```

Comment:
seems to be an incorrect pattern and missing max length constraint 

https://docs.aws.amazon.com/iot/latest/apireference/API_AttributePayload.html